Issue #545 - Added NamingService, responsible of providing the...

mathieucarbou opened the pull request:
- PR for issue #545 
- Depends on Issue #540 (solved by PR #551)

The idea here is to provide a service, configurable by a user, which is responsible for naming the cache manager this service is used into.

The `NamingService` would thus be required by the management service, which need to have a cache manager name. Other services requiring a name would also be able to depend on this service. 

Naming a cache manager is optional, and just needed by some services.

Note:

This PR contains 2 commits from PR #551 and 1 commit for the NamingService (20969f2)

albinsuresh commented:
Shouldn't there be some documentation on this in the GettingStarted doc and samples, since this is a user facing API?

mathieucarbou commented:
I think so also. 
Actually this pull request is to introduce a more bigger refactoring of the DefaultManagementRegistry (#552) which will use this service for naming the cache managers. 
And before adding documentation and so on, I was much more thinking of preparing all pull requests then discuss with the team all the pull request + DefaultManagementRegistry refactoring.
And if all is good I will update pull requests with the reviews :-)
Does it work for you guys ? 
@ljacomet @lorban @albinsuresh  ?

mathieucarbou commented:
As per @lorban suggestions:

- Renaming from *Name* to *Alias*
- Added `String AliasService.getCacheAlias(Cache)`

Also:

- Added tests
- Added more javadoc to precise that the implementation could change

mathieucarbou commented:
update: introducing `CacheBinding` so that listeners can be aware of cache alias at cache init and close. This allows to decouple the alias service from the static classes StatisticsManager and ContextManager used to store the alias of cache and cache managers.

mathieucarbou commented:
test this please

mathieucarbou commented:
will create a pull request with several separated commits
